kong/utils.go

```diff
@@ -327,6 +368,63 @@ func fillConfigRecord(schema gjson.Result, config Configuration) Configuration {
 		return true
 	})
 
+	// Filling defaults for deprecated fields
+	// Required for deck sync/diff inorder
+	// Otherwise, users keep seeing updates in these fields despite of no change
+	shortHandFields.ForEach(func(_, value gjson.Result) bool {
+		ms := value.Map()
+		fname := ""
+		for k := range ms {
+			fname = k
+			break
+		}
+
+		var deprecatedFieldValue interface{}
+
+		// check if key is already set in the config
+		if v, ok := config[fname]; ok {
+			if v != nil {
+				// This config's value should be retained.
+				// Also, the resul config 'res' may have a different value for some nested fields than this.
+				// As per current conventions, shorthand fields take priority when different values are present
+				// in equivalent shorthand configs and normal nested configs.
+				// Backfilling nested configs to reduce inconsistencies.
+				deprecatedFieldValue = v
+			}
+		}
+
+		// Using path provided in backwards translation to get
+		// the defaults for deprecated fields from the already formed default config
+		backwardTranslation := value.Get(fname + ".translate_backwards")
+
+		if !backwardTranslation.Exists() {
+			// This block attempts to fill defaults for deprecated fields.
+			// Thus, not erroring out here, as it is not vital.
+			return true
+		}
+
+		var configPathForBackwardTranslation []string
+		for _, value := range backwardTranslation.Array() {
+			configPathForBackwardTranslation = append(configPathForBackwardTranslation, value.Str)
+		}
+
+		if deprecatedFieldValue != nil {
+			_ = backfillResultConfigMap(res, configPathForBackwardTranslation, deprecatedFieldValue)
```

Comment:
Why not check the error here? 🤔

Reply:
Similar reason to why we are not erroring out for other cases. This entire block attempts to fill values/defaults for deprecated fields. Thus, it doesn't seem so crucial to error out.
If we let the thing pass, deck sync or deck diff may show some updates, but the process will still go through.

Adding a comment in the code to explain the same.